types/vue.d.ts

```diff
@@ -42,6 +42,7 @@ export declare class Vue {
   readonly $slots: { [key: string]: VNode[] };
   readonly $scopedSlots: { [key: string]: ScopedSlot };
   readonly $isServer: boolean;
+  $props: Object;
```

Comment:
In TypeScript, it would be better to use `any` rather than `Object` because the latter does not allow property access as it indicates super class of all object type (not *any* object type).
e.g. We cannot write `this.$props.foo` even if we define `foo` props.

In addition, can we add `readonly` modifier since `$props` is readonly property on userland? 🙂